include/deal.II/base/tensor.h

```diff
@@ -1094,9 +1110,14 @@ namespace internal
       return values[i];
     }
 
+    template <typename ArrayElementType>
+    struct Uninitialized
+    {
+      static ArrayElementType value;
+    };
```

Comment:
Originally, the following function definition existed.
```c++
     template <typename ArrayElementType>
     ArrayElementType &
     subscript(ArrayElementType *,
               const unsigned int,
               std::integral_constant<int, 0>)
     {
       Assert(
         false,
         ExcMessage(
           "Cannot access elements of an object of type Tensor<rank,0,Number>."));
       static ArrayElementType t;
       return t;
     }
```
The static variable defined within the function did now allow constexpr evaluation of the function (see [here](https://godbolt.org/z/ftsoMC)). So I made a workaround by creating `struct Uninitialized` outside the function within the `internal::TensorSubscriptor` namespace.

Reply:
If this is the case, then your workaround seems reasonable to me. It would be helpful just to document in the code why the struct is necessary. Perhaps a slight permutation of exactly what you've written here, placed as a normal comment just above the struct declaration, would suffice :-)

Reply:
Added the explanation comment in my latest commit